core/src/legacy/legacy-handlers.ts

```diff
@@ -0,0 +1,43 @@
+import type {
+  CapacitorInstance,
+  WindowCapacitor,
+} from '../definitions-internal';
+import { noop } from '../util';
+
+export const initLegacyHandlers = (
+  win: WindowCapacitor,
+  cap: CapacitorInstance,
+): void => {
+  // define cordova if it's not there already
+  win.cordova = win.cordova || {};
+
+  const doc = win.document;
+  const nav = win.navigator;
+
+  if (nav) {
+    nav.app = nav.app || {};
+    nav.app.exitApp = () => {
+      cap.nativeCallback('App', 'exitApp', {});
+    };
+  }
+
+  if (doc) {
+    const docAddEventListener = doc.addEventListener;
+    doc.addEventListener = (...args: any[]) => {
+      const eventName = args[0];
+      const handler = args[1];
+      if (eventName === 'deviceready' && handler) {
+        Promise.resolve(handler);
```

Comment:
Is this a typo? I'm pretty sure this line does nothing.

Reply:
yup, fixed thanks https://github.com/ionic-team/capacitor/commit/da7b796e09d20294d7dc717b72f238d912b960e2